app/reducers/index.js

```diff
@@ -1,19 +1,33 @@
 import { routerReducer as routing } from 'react-router-redux';
 import { combineReducers } from 'redux';
-import * as types from '../actions/types.js';
+import * as types from '../actions/types';
 
-const filter = (state = '', action) => {
+const initialState = {
+    geolocation: null,
+    city: null,
+    weather: null,
+};
+
+function weatherApp(state = initialState, action) {
     switch (action.type) {
-        case types.FILTER:
-            return action.filter
+        case types.SET_CITY:
+            return {
+                city: action.city,
+                ...state,
+            };
+        case types.UPDATE_WEATHER_INFO:
+            return {
+                weather: action.weather,
+                ...state,
```

Comment:
it's a point in suggested requirements, so default eslint settings demand this comma